components/ILIAS/Contact/BuddySystem/classes/Tables/RelationsTable.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\Contact\BuddySystem\Tables;
+
+use ILIAS\Contact\TableRows;
+use ILIAS\UI\Component\Table\DataRow;
+use ilBuddySystemArrayCollection;
+use ILIAS\UI\Component\Component;
+use ilBuddySystemRelationState;
+use ILIAS\UI\Component\Table\Action\Action;
+use ilCtrlInterface;
+use ILIAS\UI\Component\Table\DataRowBuilder;
+use ILIAS\Data\Range;
+use ILIAS\Data\Order;
+use ilObjUser;
+use ilStr;
+use ilUserUtil;
+use ilBuddyList;
+use ilBuddySystemRelationStateFactory;
+use ilBuddySystemRelation;
+use ilLanguage;
+use ILIAS\UI\Factory as UIFactory;
+use ilUIService;
+use ILIAS\UI\Component\Input\Container\Filter\Standard as Filter;
+use Closure;
+use ILIAS\HTTP\GlobalHttpState as Http;
+use Generator;
+
+class RelationsTable
+{
+    public function __construct(
+        private readonly UIFactory $create,
+        private readonly ilLanguage $lng,
+        private readonly ilUIService $ui_service,
+        private readonly Http $http
+    ) {
+    }
+
+    /**
+     * @param array{state?: string, name?: string} $filter
+     * @return array<array{user_id: int, public_name: string, login: string, state: ilBuddySystemRelationState, points: ilBuddySystemRelationState[], state-text: string}>
```

Comment:
Why did you choose the term `points` :)`

Reply:
Like from point A to point B.